src/TACSBuckling.cpp

```diff
@@ -239,6 +239,10 @@ void TACSLinearBuckling::solve(TACSVec *rhs, TACSVec *u0, KSMPrint *ksm_print) {
       assembler->applyBCs(path);
     } else {
       pc->factor();
+
+      // set BCs associated with displacements (displacement control)
+      assembler->setBCs(path);
+      assembler->setVariables(path);
```

Comment:
Lines 244-245 are the offending lines that when commented out => the adjoint is correct again and the derivative tests pass.

Reply:
Are these lines actually needed? It seems like the solve should run fine w/o it

Reply:
The nonzero displacement constraints do not work for the buckling without these lines. Especially for shear buckling

Reply:
They should. This is only changing the initial guess for the linear that happens in 256. Which means these values get overwritten anyways

Reply:
The update is probably not being applied correctly after the solve happens

Reply:
Ok I'll check it out again

Reply:
@sean-engelstad have a look at this fix: https://github.com/sean-engelstad/tacs/pull/1